examples/router_resource.rs

```diff
@@ -0,0 +1,65 @@
+//! Example: Updating components with use_resource
+//! -----------------
+//!
+//! This example shows how to use use_reactive to update a component properly
+//! when linking to it from the same component, when using use_resource
+
+use dioxus::prelude::*;
+
+#[derive(Clone, Routable, Debug, PartialEq)]
+enum Route {
+    #[route("/")]
+    Home {},
+    #[route("/blog/:id")]
+    Blog { id: i32 },
+}
+
+fn main() {
+    launch(App);
+}
+
+#[component]
+fn App() -> Element {
+    rsx! {
+        Router::<Route> {}
+    }
+}
+
+#[component]
+fn Blog(id: i32) -> Element {
+    async fn future(n: i32) -> i32 {
+        n
+    }
+
+    // if you use the naive approach, the "Blog post {id}" below will never update when clicking links!
+    // let res = use_resource(move || future(id));
+
+    // the use_reactive hook is required to properly update when clicking links to this component, from this component
+    let res = use_resource(use_reactive(&id, |id| future(id)));
+
+    match res() {
```

Comment:
Using `res()` here is recommended over `res.read_unchecked()` because the result of the resource is very cheap to clone. `read_unchecked` should be preferred when you don't want to clone the data

Reply:
That sounds very much like something worth documenting! Would https://dioxuslabs.com/learn/0.5/reference/use_resource be the right place, or is there somewhere else better, in your mind?